task2/main.c

```diff
@@ -0,0 +1,102 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+#include "stack.h"
+
+void printFunctions()
+{
+    printf("Functions:\n");
+    printf("0 - exit\n");
+    printf("1 - push\n");
+    printf("2 - pop\n");
+    printf("3 - add\n");
+    printf("4 - print stack\n");
+    printf("5 - print list of functions\n");
+}
+
+int main()
+{
+    Stack *stack = NULL;
+    if (stackCreate(&stack) == MemoryAllocationError)
+    {
+        printf("Memory allocation error!\n");
+        return MemoryAllocationError;
+    }
+
+    printf("---STACK---\n");
+    printFunctions();
+    bool exitFlag = false;
+    int errorCode = 0;
+    while (!exitFlag)
+    {
+        int scanResult = 0;
+        int function = -1;
+        while (!scanResult || !(function >= 0 && function <= 5))
+        {
+            printf("Enter the function: ");
+            scanResult = scanf("%d", &function);
+            if (!scanResult || !(function >= 0 && function <= 5))
+            {
+                printf("Incorrect input!\n");
+                scanf("%*[^\n]");
+            }
+        }
+
+        switch (function)
+        {
+        case 0:
+            printf("Bye!\n");
+            exitFlag = true;
+            break;
+        case 1:
+            int value = 0;
+            scanResult = 0;
+            while (!scanResult)
+            {
+                printf("Enter the value: ");
+                scanResult = scanf("%d", &value);
+                if (!scanResult)
+                {
+                    printf("Incorrect input!\n");
+                    scanf("%d*[^\n]");
```

Comment:
```suggestion
                    scanf("%*[^\n]");
```
Откуда здесь символ `d`? О_о
(-0.1 балла)